### DISCUSSION TITLE
Stale connection not removed after cable update

### DISCUSSION THREAD

**Discussion by tfboy** [netbox-community/netbox#12930] [Help Wanted!]
First time I've seen this in a long time.
Running Netbox 3.5.1
I had a cable connecting one device to the other (server NIC to a network switch port).
I edited the cable as I needed to move the port it's connected to on the switch side.
The switch now shows the new port number with the cable number and the "S" trace button enabled.
But the old port still shows the "connection data". Cable and trace functions are gone.
![image](https://github.com/netbox-community/netbox/assets/6972726/0d7c25f2-5245-4a03-b812-bdf41eabf661)

I've tried a reboot and run a mange.py healthcheck and I think everything is OK.
```
python3 netbox/manage.py housekeeping
[*] Clearing expired authentication sessions
        Sessions cleared.
[*] Checking for expired changelog records
        No expired records found.
[*] Checking for expired jobs
        No expired records found.
[*] Checking for latest release
        Skipping: RELEASE_CHECK_URL not set
Finished.
```
I did recently play around with this switch as it's part of a virtual chassis. I don't know if the updates have caused some kind of corruption or not cleared a field meaning this connection seems to remain.

I can't figure out how to clear it!

Any ideas?

**Comment by mtinberg** [comment]
I don't know if this is a possible problem but the cable traces are cached, maybe this has gotten out of sync somehow.  Does running netbox/manage.py trace_paths make any change to the behavior?  If this fixes it then you can troubleshoot how the update trigger didn't get fired or failed, and if this doesn't fix it the issue lies elsewhere.

sudo -u netbox /opt/netbox/current/venv-py3/bin/python /opt/netbox/current/netbox/manage.py trace_paths --no-input


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
From: tfboy ***@***.***>
Sent: Monday, June 19, 2023 4:33 AM
To: netbox-community/netbox ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [netbox-community/netbox] Stale connection not removed after cable update (Discussion #12930)


First time I've seen this in a long time.
Running Netbox 3.5.1
I had a cable connecting one device to the other (server NIC to a network switch port).
I edited the cable as I needed to move the port it's connected to on the switch side.
The switch now shows the new port number with the cable number and the "S" trace button enabled.
But the old port still shows the "connection data". Cable and trace functions are gone.
[image]<https://user-images.githubusercontent.com/6972726/246788996-0d7c25f2-5245-4a03-b812-bdf41eabf661.png>

I've tried a reboot and run a mange.py healthcheck and I think everything is OK.

python3 netbox/manage.py housekeeping
[*] Clearing expired authentication sessions
        Sessions cleared.
[*] Checking for expired changelog records
        No expired records found.
[*] Checking for expired jobs
        No expired records found.
[*] Checking for latest release
        Skipping: RELEASE_CHECK_URL not set
Finished.


I did recently play around with this switch as it's part of a virtual chassis. I don't know if the updates have caused some kind of corruption or not cleared a field meaning this connection seems to remain.

I can't figure out how to clear it!

Any ideas?

**Reply by tfboy** [reply]
Hi Mike.
Thanks for the thought - I was wondering whether there was a specific caching thing to try that would have done more than a reboot.
It doesn't appear to have changed anything though:
```
$ python3 netbox/manage.py trace_paths --no-input
Found no missing console port paths; skipping
Found no missing console server port paths; skipping
Found no missing interface paths; skipping
Found no missing power feed paths; skipping
Found no missing power outlet paths; skipping
Found no missing power port paths; skipping
Finished.
```
and the old data persists :(
